test/integration/smoke/test_migration.py

```diff
@@ -76,66 +77,68 @@ def setUpClass(cls):
         cls.test_data = cls.testClient.getParsedTestDataConfig()
         cls.services = Services().services
 
+        cls.hypervisorNotSupported = False
         hypervisor = get_hypervisor_type(cls.api_client)
         if hypervisor.lower() not in ["vmware", "kvm"]:
-            raise unittest.SkipTest("This feature is supported "
-                                    "only on Vmware and KVM")
-
-    # Get Domain, Zone, Template
-        cls.domain = get_domain(cls.api_client)
-        cls.zone = get_zone(
-                cls.api_client,
-                cls.testClient.getZoneForTests())
-        cls.template = get_template(
-                cls.api_client,
-                cls.zone.id,
-                cls.test_data["ostype"]
-        )
-        cls.services["virtual_machine"]["template"] = cls.template.id
-        if cls.zone.localstorageenabled:
-            cls.storagetype = 'local'
-            cls.test_data["service_offerings"][
-                "tiny"]["storagetype"] = 'local'
-        else:
-            cls.storagetype = 'shared'
-            cls.test_data["service_offerings"][
-                "tiny"]["storagetype"] = 'shared'
-
-        cls.service_offering = ServiceOffering.create(
-                cls.api_client,
-                cls.test_data["service_offerings"]["tiny"]
-        )
+            cls.hypervisorNotSupported = True
 
-        # Create Network offering without userdata
-        cls.network_offering_nouserdata = NetworkOffering.create(
-                cls.api_client,
-                cls.test_data["network_offering"]
-        )
-        # Enable Network offering
-        cls.network_offering_nouserdata.update(cls.api_client,
-                                               state='Enabled')
-
-        # Create Network Offering with all the serices
-        cls.network_offering_all = NetworkOffering.create(
-                cls.api_client,
-                cls.test_data["isolated_network_offering"]
-        )
-        # Enable Network offering
-        cls.network_offering_all.update(cls.api_client, state='Enabled')
+        cls._cleanup = []
+        if not cls.hypervisorNotSupported:
```

Comment:
a double negation. can we reverse the meaning (and name) of the variable?

Reply:
I would agree but would need to check if the annotation @skipTestIf accepts negations